language/move-lang/src/shared/mod.rs

```diff
@@ -197,6 +199,51 @@ pub fn shortest_cycle<'a, T: Ord + Hash>(
     path
 }
 
+//**************************************************************************************************
+// Compilation Env
+//**************************************************************************************************
+
+pub struct CompilationEnv {
+    flags: Flags,
+    errors: Errors,
+    // TODO(tzakian): Remove the global counter and use this counter instead
+    // pub counter: u64,
+}
+
+impl CompilationEnv {
+    pub fn new(flags: Flags) -> Self {
+        Self {
+            flags,
+            errors: Vec::new(),
+        }
+    }
+
+    pub fn add_error(&mut self, e: Vec<(Loc, impl Into<String>)>) {
+        self.errors
+            .push(e.into_iter().map(|(loc, msg)| (loc, msg.into())).collect())
+    }
+
+    pub fn add_errors(&mut self, es: Errors) {
+        self.errors.extend(es)
+    }
+
+    pub fn has_errors(&self) -> bool {
+        !self.errors.is_empty()
+    }
+
+    pub fn check_errors(&mut self) -> Result<(), Errors> {
+        if self.has_errors() {
+            Err(std::mem::take(&mut self.errors))
+        } else {
+            Ok(())
+        }
+    }
+
+    pub fn get_flags(&self) -> &Flags {
```

Comment:
Nit I would still remove the `get_` prefix. It does not feel rusty to me. Typically "getters" just have the same name as the field, at least as I have been writing them. Unless the `get_` is doing some computation